app/models/git_repository.rb

```diff
@@ -114,6 +114,27 @@ def file_changed?(sha1, sha2, file)
     executor.execute!("cd #{pwd}", "git diff --quiet --name-only #{sha1}..#{sha2} #{file}")
   end
 
+  def directory_contents(git_ref, path: '.')
+    cmd = "git ls-tree -r #{git_ref} --name-only --full-tree #{path}"
+    valid, output = run_single_command(cmd) { |line|
+      if block_given?
+        yield line.chomp
+      else
+        line.chomp
+      end
+    }
+    Rails.logger.error("Could not list the files in directory '#{path}'. Git Ref '#{git_ref}'") unless valid
+    output
+  end
+
+  def file_contents(git_ref, file_path)
+    #first command disables less as default pager
+    cmds = ["git config core.pager ''", "git show -q #{git_ref}:#{file_path}"]
+    valid, output = run_commands(cmds) { |line| line }
+    Rails.logger.error("Could not retrieve file contents for '#{file_path}'. Git Ref '#{git_ref}'") unless valid
+    output.join
+  end
```

Comment:
Technically, you could have code that looks like this:

```ruby
repo = GitRepository.new(repository_url: 'github.com/foo/bar', repository_dir: '/tmp/foo')
repo.file_contents('master', 'README.md')
```

In that scenario, `setup!` wasn't called, so the local cache doesn't even exist. I'm guessing it would fail somewhere in `run_commands` with a cryptic error message.

Maybe add a check at the beginning of the method like `fail "Not setup" unless locally_cached?`